### ISSUE TITLE
[Request] "Moveable" PROGRESS BAR position

### ISSUE THREAD

**Issue by bolsoncerrado** [jneilliii/OctoPrint-PrusaSlicerThumbnails#113]
https://github.com/OctoPrint/OctoPrint/issues/4846

**Is your feature request related to a problem? Please describe.**

I run a farm where I open multiple Octoprint tabs in a single browser.

With the recent addition of slicer thumbnails I have to scrolldown past the thumbnails to find out a "quick way" to see the print progress of each printer.

**Describe the solution you'd like**

It would be great a setting to determine the position of the PROGRESS BAR, for example BEFORE the thumbnails would be great!

**Describe alternatives you've considered**

Tried to look for plugins but no joy.

**Additional alternatives**

Perhaps adding a VERTICAL progress bar sideways with the thumbnail like "Prusaconnect" does? :P 

<img width="357" alt="Captura de pantalla 2023-07-07 a las 16 28 09" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/7b9768a0-95ac-4eb2-a43e-60242f8e6e5c">

**Comment by jneilliii** [comment]
I won't get into flipping to vertical or anything like that, but did a quick test in developer tools of the browser and think this might be a plausible option...

![image](https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/5249455/9a5c58d1-c346-4476-9b48-ee8d3decae53)

to test yourself, open browser's developer tools (F12 on keyboard typically) and in the console tab copy/paste this and hit enter.

```
$('#state > div > div.progress.progress-text-centered').css({'margin-bottom': 'inherit'}).insertBefore('#prusaslicer_state_thumbnail').after('<hr>');
```

**Comment by jneilliii** [comment]
another alternative would be this plugin, and it would put the progress in the title of the browser.

https://plugins.octoprint.org/plugins/progress_title/

**Comment by bolsoncerrado** [comment]
> I won't get into flipping to vertical or anything like that, but did a quick test in developer tools of the browser and think this might be a plausible option...

You mean you  could code this permanent and add a setting option? If so, that could do for me too.

The progress title one is not as visual since it's text based... ;)

**Comment by bolsoncerrado** [comment]
<img width="311" alt="Captura de pantalla 2023-07-07 a las 16 44 02" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/1bbd20f0-e4c7-445f-aa1a-712aa86c4a6c">


Looks cool indeed! ;)

**Comment by jneilliii** [comment]
> You mean you could code this permanent and add a setting option?

Yes, this is the idea. I just wanted to confirm before putting the work in to make this change that it was acceptable.

**Comment by jneilliii** [comment]
Added in above commit and released as version 1.0.3rc1. Switch the plugin's release channel in OctoPrint's Software Update settings to `Release Candidate` and update when prompted.

**Comment by bolsoncerrado** [comment]
Thanks!

**Comment by bolsoncerrado** [comment]
And now that you're at it what about adding a second HR after the thumbnail for "politeness" view ;)

Before:

<img width="297" alt="Captura de pantalla 2023-07-08 a las 19 52 36" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/d75fa5ea-2af4-41c1-9bf2-8730748a9067">


After:

<img width="297" alt="after" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/abc98672-4a54-4ed7-818d-d9cc6500ce07">



:)

**Comment by jneilliii** [comment]
I could do that for images that are 100% only, because with UI Customizer and a wider left panel with scaled down image would mess up with the hr. 

![image](https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/5249455/38c346bc-c091-4884-8114-a04ff8e411a7)

**Comment by bolsoncerrado** [comment]
If you mean this

<img width="240" alt="Captura de pantalla 2023-07-08 a las 20 20 50" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/ba2f596b-4738-457e-9451-1fe18954f783">

then it works for me ;)

**Comment by jneilliii** [comment]
added in the above commit and will be release as version 1.0.3rc2 shortly. you can do a force check for updates from OctoPrint's Software Update settings at the top.

![image](https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/5249455/215db328-6a37-4f95-b1ac-8d98f84d466e)

**Comment by bolsoncerrado** [comment]
Gorgeous!! 

👍 

<img width="323" alt="Captura de pantalla 2023-07-09 a las 10 48 08" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/8fac01ac-7cea-4bab-bf45-0dc4ed108495">

**Comment by bolsoncerrado** [comment]
Reopening due to "minor issue":

Scenario: a pre-existing file in the NAS drive where all instances have "visible" thumbnail at sight, I just pressed PRINT on each instance, and on one of them, the progress bar didn't load on top.  I know refreshing the browser will fix the issue, I just found "interesting" that all the instances are already upgraded to the latest RC with the additional HR and all  browsers refreshed prior to start this print. In fact I had to turn on the machines as I already turned them off yesterday.

What a weirdo. Must be something related to the "speed" the image is parsed vs the progress bar code is filled? IDK.

<img width="1888" alt="Captura de pantalla 2023-07-09 a las 17 38 49" src="https://github.com/jneilliii/OctoPrint-PrusaSlicerThumbnails/assets/10992914/32d7836d-6d0e-4141-bbb3-79a7db9feb96">
